### ISSUE TITLE
enable offline running

### ISSUE THREAD

**Issue by djschny** [kubernetes-sigs/kubeadm-dind-cluster#21]
Assuming you've already ran a `./dind-cluster-v1.6.sh up` successfully once, it would be nice to allow the cluster to still start up offline with no internet connection (airplane, park, etc.).

Turn off wifi on your laptop and run `./dind-cluster-v1.6.sh up`

```
$ ./dind-cluster-v1.6.sh up
* Making sure DIND image is up to date 
Error response from daemon: Get https://registry-1.docker.io/v2/: dial tcp: lookup registry-1.docker.io on 192.168.65.1:53: read udp 192.168.65.2:43749->192.168.65.1:53: i/o timeout
```

**Comment by ivan4th** [comment]
Thanks, will try and fix this

**Comment by jellonek** [comment]
@djschny can you verify if `./dind-cluster-v1.6.sh reup` (defined [there](https://github.com/Mirantis/kubeadm-dind-cluster/blob/master/fixed/dind-cluster-v1.6.sh#L924), which is almost the same as `up`) does what you are asking for?

**Comment by jellonek** [comment]
@djschny ping.

That works for me.

**Comment by djschny** [comment]
I just tried `./dind-cluster-v1.7.sh reup` and it definitely gets further, but unfortunately it hangs on the dashboard. I observed this by waiting forever at the dashboard setup and then examined the pods:

```
$ kubectl get pods --namespace=kube-system 
NAME                                        READY     STATUS         RESTARTS   AGE
calico-etcd-4m229                           1/1       Running        0          1m
calico-node-9s9bj                           2/2       Running        21         16d
calico-node-rdvbd                           2/2       Running        20         16d
calico-node-zpcx3                           2/2       Running        19         16d
calico-policy-controller-1727037546-m95wk   1/1       Running        12         16d
etcd-kube-master                            1/1       Running        6          16d
kube-apiserver-kube-master                  1/1       Running        6          16d
kube-controller-manager-kube-master         1/1       Running        6          20s
kube-dns-2185897156-mxp7c                   3/3       Running        0          1m
kube-proxy-ljdcn                            1/1       Running        0          1m
kube-proxy-pr1ld                            1/1       Running        0          1m
kube-proxy-t9wtq                            1/1       Running        0          1m
kube-scheduler-kube-master                  1/1       Running        6          16d
kubernetes-dashboard-1136505239-k8dr4       0/1       ErrImagePull   0          1m
```

**Comment by zoobab** [comment]
I have managed to put kubeadm-dind into one big docker image (docker-in-docker-in-docker), but the image is 5GB big.

Would there be a way to reduce drastically the size of the whole cluster?

**Comment by webwurst** [comment]
@zoobab Can you share your `Dockerfile`?

**Comment by zoobab** [comment]
@webwurst I hit a problem trying to do a docker build combined with privileged mode (needed to launch the dind-cluster-v1.6.sh shell script). So my attempt to make a Dockerfile stopped there.

See:

https://github.com/moby/moby/issues/1916

Then I just made a container by hand with ubuntu, and saved the container with a docker save as an output. The only thing I had to change was the listening address on 0.0.0.0 in order to expose the ports externally.

Note that should be fairly scriptable, but not as a Dockerfile atm.
